### DISCUSSION TITLE
Implement more resampling schemes

### DISCUSSION THREAD

**Discussion by AdrienCorenflos** [blackjax-devs/blackjax#121] [New features]
1. Implement Srinivasan Sampling Process (SSP) resampling. Ref: https://arxiv.org/abs/1707.01845
2. Implement chopthin resampling. Ref: https://arxiv.org/abs/1502.07532. Note this cannot (directly?) be used with waste-free SMC
3. Implement Metropolis (and rejection perhaps) resampling. Ref: Parallel resampling in the particle filter. Note, the Metropolis resampler produces a biased sample.
4. Verify if non-exchangeable resampler can be used withing residual resampling, if so make it work.
5. Implement OT resampling.

**Comment by rlouf** [comment]
I guess that for now SSP would make the most sense?

**Reply by AdrienCorenflos** [reply]
I'd say SSP and metropolis yeah. Chopthin would require returning modified weights after resampling instead of assuming they are uniform.

**Comment by junpenglao** [comment]
🤖 *This reply was drafted with Claude and reviewed by me (@junpenglao).*

Status: `systematic`, `stratified`, `multinomial`, and `residual` resampling are all implemented in `blackjax/smc/resampling.py`. Still open from this list: SSP, chopthin, and Metropolis/rejection resampling. PRs welcome for those.
